src/main/fc/core.c

```diff
@@ -495,7 +496,8 @@ void tryArm(void)
         const timeUs_t currentTimeUs = micros();
 
 #ifdef USE_DSHOT
-        if (currentTimeUs - getLastDshotBeaconCommandTimeUs() < DSHOT_BEACON_GUARD_DELAY_US) {
+        if (currentTimeUs - getLastDshotBeaconCommandTimeUs() < DSHOT_BEACON_GUARD_DELAY_US 
+        || (currentTimeUs - lastDisarmTimeUs < 200000 && flipOverAfterCrashWasActiveLast)) {
```

Comment:
This logic should be moved to line 528

Reply:
Would you mind elaborating on why we would want it on line 528?

Reply:
Yeah probably mean 536 as there we handle `isModeActivationConditionPresent(BOXFLIPOVERAFTERCRASH)`

Reply:
So my justification for putting it here is because the beacon guard delay is the exact same function as I'm trying to achieve here. I want arming to be delayed if crashflip was turned off too soon. Not sure the inner workings of it on the esc side, but when the motors are commanded to switch direction, there needs to be delay, otherwise the motors will not respond to the command and will spin the wrong way, causing a runaway takeoff.

Reply:
Understand the intention. Also not a request but a comment. Just trying to move this case to where it's supposed to be handled. Happy to have comment from other reviewers.

Reply:
Maybe put a comment in explaining the intent>

Reply:
The actual technical requirement is not that some arbitrary `flipOverAfterCrashWasActiveLast` has expired, but a sequence of DSHOT commands + a delay period has elapsed.

The DSHOT motor subsystem should maintain calculate `commandDelayExpiresAt` value based on the operations being performed and then generic code, that is not specifically related to crash-flip, should be used which uses the `commandDelayExpiresAt` appropriately.

Reply:
in addition to my previous comment above, perhaps a `dshot_isReady()` call, and implementation, would be far more appropriate than some arbitrary delay.